src/joy_api.c

```diff
@@ -76,12 +80,249 @@ struct configuration active_config;
+ *
+ * Returns:
+ *      formatted data length is returned
+ *
+ */
+static unsigned int joy_splt_format_data(flow_record_t *rec,
+                                         JOY_EXPORT_TYPE export_frmt,
+                                         unsigned char *data)
+{
+    unsigned int i = 0;
+    unsigned int entries_used = 0;
+    unsigned int num_of_pkts = 0;
+    unsigned int data_len = 0;
+    struct timeval ts;
+    short *formatted_data = (short*)data;
+
+    /* see how many packets we have to process - max is MAX_NFV9_SPLT_SALT_PKTS */
+    num_of_pkts = (rec->op < MAX_NFV9_SPLT_SALT_PKTS) ? rec->op : MAX_NFV9_SPLT_SALT_PKTS;
+
+    /* figure out the length of the data we are formatting */
+    if (export_frmt == JOY_NFV9_EXPORT) {
+        /* NFv9 is always 40 bytes */
+        data_len = MAX_NFV9_SPLT_SALT_ARRAY_LENGTH;
+    } else {
+        /* IPFix is variable length - each entry is represented by 2 16-bit values */
+        data_len = num_of_pkts * 4;
+    }
+
+    /* format for Netflow v9 export */
+    if (export_frmt == JOY_NFV9_EXPORT) {
+        /* loop through the SPLT lengths and store appropriately */
+        for (i=0; i < num_of_pkts; ++i) {
+            *(formatted_data+i) = (short)rec->pkt_len[i];
+        }
+
+        if (num_of_pkts < MAX_NFV9_SPLT_SALT_PKTS) {
+            /* padding needs to occur */
+            for (;i < MAX_NFV9_SPLT_SALT_PKTS; ++i) {
+               *(formatted_data+i) = (short)-32768;
+            }
+        }
+
+        /* loop through the SPLT times and store appropriately */
+        for (i=0; i < num_of_pkts; ++i) {
+            if (i > 0) {
+                joy_timer_sub(&rec->pkt_time[i], &rec->pkt_time[i-1], &ts);
+            } else {
+                joy_timer_sub(&rec->pkt_time[i], &rec->start, &ts);
+            }
+            *(formatted_data+MAX_NFV9_SPLT_SALT_PKTS+i) =
+                 (short)joy_timeval_to_milliseconds(ts);
+        }
+
+        if (num_of_pkts < MAX_NFV9_SPLT_SALT_PKTS) {
+            /* padding needs to occur */
+            for (;i < MAX_NFV9_SPLT_SALT_PKTS; ++ i) {
+               *(formatted_data+MAX_NFV9_SPLT_SALT_PKTS+i) = (short)0x00;
+            }
+        }
+
+    /* else format for IPFix export */
+    } else {
+        /* loop through the SPLT lengths and store appropriately */
+        for (i=0; i < num_of_pkts; ++i) {
+            *(formatted_data+i) = (short)rec->pkt_len[i];
+        }
+
+        /* store how many entries we used since IPFix doesn't pad */
+        entries_used = i;
+
+        /* loop through the SPLT times and store appropriately */
+        for (i=0; i < num_of_pkts; ++i) {
+            if (i > 0) {
+                joy_timer_sub(&rec->pkt_time[i], &rec->pkt_time[i-1], &ts);
+            } else {
+                joy_timer_sub(&rec->pkt_time[i], &rec->start, &ts);
+            }
+            *(formatted_data+entries_used+i) =
+                 (short)joy_timeval_to_milliseconds(ts);
```

Comment:
We should be consistent on types. uint16_t

Reply:
probably